sdk/storage/azure-storage-blob/azure/storage/blob/_download.py

```diff
@@ -205,17 +204,17 @@ def _download_chunk(self, chunk_start, chunk_end):
                         download_stream_current=self.progress_total,
                         **self.request_options
                     )
-                except HttpResponseError as error:
-                    process_storage_error(error)
-
-                try:
-                    chunk_data = process_content(response, offset[0], offset[1], self.encryption_options)
                     retry_active = False
-                except (requests.exceptions.ChunkedEncodingError, requests.exceptions.ConnectionError) as error:
+                except (IncompleteReadError, ServiceResponseError) as error:
```

Comment:
The requests transport will catch `requests.exceptions.ConnectionError` and raise it as a `ServiceResponseError` ([here](/sdk/core/azure-core/azure/core/pipeline/transport/_requests_basic.py:34))